src/Geometry/CSG/CSG.jl

```diff
@@ -126,12 +128,14 @@ Create a (pseudo) leaf node from another CSGGenerator, this is useful if you wan
 function leaf(n::CSGGenerator{T}, transform::Transform{T} = identitytransform(T))::CSGGenerator{T} where {T<:Real}
     return CSGGenerator{T}((parenttransform) -> n(parenttransform * transform))
 end
-export leaf
 
 # CSG objects are trees, not graphs, since it makes no sense to reuse intermediate results. Hence no need to keep track of common subexpressions.
 # Static arrays are much faster than mutable arrays so want node transforms to be static. Unfortunately the transformations cascade from the root to the leaves but the nodes have to be created from the leaves to the roots.
 # Wrap the csg operations in function that delays evaluation until the transform has been computed.
 
+!(a::CSGGenerator{T}) where {T<:Real} = CSGGenerator{T}((parenttransform) -> ComplementNode(a(parenttransform)))
```

Comment:
We don't want to expose this as it will allow users to construct CSG trees that may not conform the assumption we make for bounding box culling of rays - users should only use intersection, union and difference

Reply:
good point, i'll remove it